composer/spec/fixtures/projects/platform_package/composer.json

```diff
@@ -0,0 +1,13 @@
+{
+  "name": "valid/name",
+  "version": "3.1.7",
+  "dist": {
+    "url": "https://www.example.net/files/3.1.7.zip",
+    "type": "zip"
+  },
+  "require": {
+    "monolog/monolog" : "1.0.1",
+    "symfony/polyfill-mbstring": "1.0.1",
+    "php": "*"
```

Comment:
How about requiring some more platform packages to exhaust the different options like `lib-*`, `ext-*` and so on?

Reply:
The regex is already tested thoroughly in composer itself and we short circuit the logic most of the time by checking the lockfile, but I've added a bit more coverage in the helper tests